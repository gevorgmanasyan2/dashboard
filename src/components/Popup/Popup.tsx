import React, { FC, ReactElement, ReactNode } from 'react'
import Modal from 'antd/lib/modal/Modal'

interface PopupProps {
  className: string
  modalTitle: string
  okButtonText?: string | ReactElement
  cancelButtonText?: string
  visible?: boolean
  closeModal?: () => void
  footer?: null | ReactNode
  mask?: boolean
}

const Popup: FC<PopupProps> = ({
  className,
  visible,
  closeModal,
  children,
  footer,
  modalTitle,
  okButtonText,
  cancelButtonText,
  mask,
}) => {
  return (
    <Modal
      title={modalTitle}
      visible={visible}
      footer={footer}
      centered={false}
      mask={mask}
      onCancel={closeModal}
      className={className}
      okText={okButtonText}
      cancelText={cancelButtonText}
    >
      {children}
    </Modal>
  )
}

export default Popup
