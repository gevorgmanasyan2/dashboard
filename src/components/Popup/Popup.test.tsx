/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import Popup from './Popup'

it('should render Popup component', () => {
  const mockItem = {
    className: 'modal',
    visible: false,
    children: <div></div>,
    footer: null,
    modalTitle: 'title',
    okButtonText:'save',
    cancelButtonText:'cancel',
    closeModal: () => {}
  }
  expect(shallow(<Popup {...mockItem} />)).toMatchSnapshot()
})
