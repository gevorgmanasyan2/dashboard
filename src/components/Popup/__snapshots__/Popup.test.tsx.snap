// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Popup component 1`] = `
<Modal
  cancelText="cancel"
  centered={false}
  className="modal"
  confirmLoading={false}
  footer={null}
  maskTransitionName="fade"
  okText="save"
  okType="primary"
  onCancel={[Function]}
  title="title"
  transitionName="zoom"
  visible={false}
  width={520}
>
  <div />
</Modal>
`;
